64_EmulatedFloatTest/main.cpp

```diff
@@ -1033,6 +1047,292 @@ class CompatibilityTest final : public MonoDeviceApplication, public MonoAssetMa
+
+                nbl::video::IGPUDescriptorSetLayout::SBinding bindings[1] = {
+                    {
+                        .binding = 0,
+                        .type = nbl::asset::IDescriptor::E_TYPE::ET_STORAGE_BUFFER,
+                        .createFlags = IGPUDescriptorSetLayout::SBinding::E_CREATE_FLAGS::ECF_NONE,
+                        .stageFlags = ShaderStage::ESS_COMPUTE,
+                        .count = 1
+                    }
+                };
+                smart_refctd_ptr<IGPUDescriptorSetLayout> dsLayout = base.m_device->createDescriptorSetLayout(bindings);
+                if (!dsLayout)
+                    base.logFail("Failed to create a Descriptor Layout!\n");
+
+                SPushConstantRange pushConstantRanges[] = {
+                {
+                    .stageFlags = ShaderStage::ESS_COMPUTE,
+                    .offset = 0,
+                    .size = sizeof(BenchmarkPushConstants)
+                }
+                };
+                m_pplnLayout = base.m_device->createPipelineLayout(pushConstantRanges, smart_refctd_ptr(dsLayout));
+                if (!m_pplnLayout)
+                    base.logFail("Failed to create a Pipeline Layout!\n");
+
+                {
+                    IGPUComputePipeline::SCreationParams params = {};
+                    params.layout = m_pplnLayout.get();
+                    params.shader.entryPoint = "main";
+                    params.shader.shader = shader.get();
+                    if (!base.m_device->createComputePipelines(nullptr, { &params,1 }, &m_pipeline))
+                        base.logFail("Failed to create pipelines (compile & link shaders)!\n");
+                }
+
+                // Allocate the memory
+                {
+                    static_assert(sizeof(float64_t) == sizeof(benchmark_emulated_float64_t));
+                    constexpr size_t BufferSize = sizeof(float64_t);
+
+                    nbl::video::IGPUBuffer::SCreationParams params = {};
+                    params.size = BufferSize;
+                    params.usage = IGPUBuffer::EUF_STORAGE_BUFFER_BIT;
+                    smart_refctd_ptr<IGPUBuffer> dummyBuff = base.m_device->createBuffer(std::move(params));
+                    if (!dummyBuff)
+                        base.logFail("Failed to create a GPU Buffer of size %d!\n", params.size);
+
+                    dummyBuff->setObjectDebugName("benchmark buffer");
+
+                    nbl::video::IDeviceMemoryBacked::SDeviceMemoryRequirements reqs = dummyBuff->getMemoryReqs();
+
+                    m_allocation = base.m_device->allocate(reqs, dummyBuff.get(), nbl::video::IDeviceMemoryAllocation::EMAF_NONE);
+                    if (!m_allocation.isValid())
+                        base.logFail("Failed to allocate Device Memory compatible with our GPU Buffer!\n");
+
+                    assert(dummyBuff->getDeviceAddress() < std::numeric_limits<uint32_t>::max()); // TODO: is it even possible?
```

Comment:
buffer device address is 64bits, why are you trying to clamp it to 32? It's risky.

Reply:
oh shit, you're using byteAddressBuffer not BDA, you shouldn't be using getDeviceAddress for that, your offset is just 0 in this case 😄
I know it was for test but you could definitely crash your gpu doing that. cause it wayyy out of bound 😄